cockatrice/src/chatview.cpp

```diff
@@ -161,72 +161,115 @@ void ChatView::appendMessage(QString message, QString sender, UserLevelFlags use
+            index = std::min(bracketFirstIndex, mentionFirstIndex);
+        }
+        else if (startsWithBracket && !startsWithAtSymbol && startsWithUrl && !startsWithHighlightWord)
+        {
+            // Contains both a bracket and URL stuff
+            index = std::min(bracketFirstIndex, urlFirstIndex);
+        }
+        else if (startsWithBracket && !startsWithAtSymbol && !startsWithUrl && startsWithHighlightWord)
+        {
+            // Contains both a bracket and a word the user wants highlighted
+            index = std::min(bracketFirstIndex, highlightWordFirstIndex);
+        }
+        else if (!startsWithBracket && startsWithAtSymbol && startsWithUrl && !startsWithHighlightWord)
+        {
+            // Contains both an @ symbol and URL stuff
+            index = std::min(mentionFirstIndex, urlFirstIndex);
+        }
+        else if (!startsWithBracket && startsWithAtSymbol && !startsWithUrl && startsWithHighlightWord)
+        {
+            // Contains both an @ symbol and a word the user wants highlighted
+            index = std::min(mentionFirstIndex, highlightWordFirstIndex);
+        }
+        else if (!startsWithBracket && !startsWithAtSymbol && startsWithUrl && startsWithHighlightWord)
+        {
+            // Contains both URL stuff and a word the user wants highlighted
+            index = std::min(urlFirstIndex, highlightWordFirstIndex);
+        }
+        else if (!startsWithBracket && startsWithAtSymbol && startsWithUrl && startsWithHighlightWord)
+        {
+            // Contains an @ symbol, URL stuff, and a word the user wants highlighted
+            index = std::min(mentionFirstIndex, std::min(urlFirstIndex, highlightWordFirstIndex));
+        }
+        else if (startsWithBracket && !startsWithAtSymbol && startsWithUrl && startsWithHighlightWord)
+        {
+            // Contains a bracket, URL stuff, and a word the user wants highlighted
+            index = std::min(bracketFirstIndex, std::min(urlFirstIndex, highlightWordFirstIndex));
+        }
+        else if (startsWithBracket && startsWithAtSymbol && !startsWithUrl && startsWithHighlightWord)
+        {
+            // Contains a bracket, an @ symbol, and a word the user wants highlighted
+            index = std::min(bracketFirstIndex, std::min(mentionFirstIndex, highlightWordFirstIndex));
+        }
+        else if (startsWithBracket && startsWithAtSymbol && startsWithUrl && !startsWithHighlightWord)
+        {
+            // Contains a bracket, an @ symbol, and URL stuff
+            index = std::min(bracketFirstIndex, std::min(mentionFirstIndex, urlFirstIndex));
+        }
+        else if (startsWithBracket && startsWithAtSymbol && startsWithUrl && startsWithHighlightWord)
+        {
+            // Contains a bracket, an @ symbol, URL stuff, and a word the user wants highlighted
+            index = std::min(highlightWordFirstIndex, std::min(bracketFirstIndex, std::min(mentionFirstIndex, urlFirstIndex)));
```

Comment:
I'm hoping @ctrlaltca or someone else can assist in thinking of a new way as this is the best I can do unfortunately.